Kernel/Process.cpp

```diff
@@ -3263,9 +3263,14 @@ int Process::sys$getsockname(int sockfd, sockaddr* addr, socklen_t* addrlen)
 
     auto& socket = *description->socket();
     REQUIRE_PROMISE_FOR_SOCKET_DOMAIN(socket.domain());
-    if (!socket.get_local_address(addr, addrlen))
+
+    OwnPtr<sockaddr> addr((sockaddr*)operator new(addrlen));
```

Comment:
Let's do this with a stack buffer instead. `sockaddr_un` is the largest kind of `sockaddr` we support, so we can just create a stack buffer with size `sizeof(sockaddr_un)` and use that.

Reply:
Yea, I had thought about that...  was just try to safe guard against a new sockaddr type ever being created.
Guess that's a bridge that will need to be crossed if the project ever gets there.